### ISSUE TITLE
MockCollectionReference.id causes unexpected

### ISSUE THREAD

**Issue by lloydrichards** [atn832/fake_cloud_firestore#210]
I came across a bug while testing collectionGroup using FakeFirestore.  When calling `.id` on a MockCollectionReference it returns an unexpected exemption and crashes the test.  In my function I use the `.parent.id` in order to check which collection I'm in and in an emulator this works no problem.  But with FakeFirestore I get a crash.  

You should be able to reproduce with the following:

```
final query = _fakeFirestore.collectionGroup("Tasks").where("created_by.id", isEqualTo: "");

print(query.docs.first.e)  // 👍 
print(query.docs.first.e.reference.parent);  // 👍 
print(query.docs.first.e.reference.parent.id);  // 👎 
```

Is this something that is already being addressed?

**Comment by atn832** [comment]
Good catch! Probably not, we haven't touched collectionGroups in a long time. If you contribute a PR and I'll gladly review it.
